catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java

```diff
@@ -387,6 +406,41 @@ public CatalogFrameworkImpl(List<CatalogProvider> catalogProviders, BundleContex
         }
     }
 
+    /**
+     * Instantiates a new CatalogFrameworkImpl
+     * @param frameworkProperties - collection of properties to be set on the CatalogFramework instance
+     */
+    public CatalogFrameworkImpl(FrameworkProperties frameworkProperties) {
+        this.context = frameworkProperties.getBundleContext();
+        this.catalogProviders = frameworkProperties.getCatalogProviders();
+        if (LOGGER.isDebugEnabled()) {
+            if (this.catalogProviders != null) {
+                LOGGER.debug("catalog providers list size = " + this.catalogProviders.size());
+            } else {
+                LOGGER.debug("catalog providers list is NULL");
+            }
+        }
+
+        this.preIngest = frameworkProperties.getPreIngest();
```

Comment:
Instead of making copies, should we just keep a reference to the `frameworkProperties` object and call the getters when this class needs something? This eliminate a bunch of fields and shrink the class a little. Not that it needs shrinking or anything ;-)